pkg/api/validation/validation.go

```diff
@@ -1722,3 +1722,26 @@ func ValidateSecurityContext(sc *api.SecurityContext) errs.ValidationErrorList {
 	}
 	return allErrs
 }
+
+func ValidateThirdPartyResource(obj *api.ThirdPartyResource) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if len(obj.Name) == 0 {
+		allErrs = append(allErrs, errs.NewFieldInvalid("name", obj.Name, "name must be non-empty"))
+	}
+	versions := util.StringSet{}
+	for ix := range obj.Versions {
+		version := &obj.Versions[ix]
+		if len(version.Name) == 0 {
+			allErrs = append(allErrs, errs.NewFieldInvalid("name", version, "name can not be empty"))
```

Comment:
Right, since the `metadata.name` of the resource specifies a unique path, and we use the `metadata.name` as most of the key in etcd, we will get write conflicts if they're not.